### DISCUSSION TITLE
Edge interpolation or something

### DISCUSSION THREAD

**Discussion by gagecarto** [developmentseed/titiler#613] [Q&A] [answered]
I have a raster dataset with values 0-255. In this dataset, 0 is no data. Rendering tiles at some zoom levels results in weird values along the edge of data/no data. 

See below where white is no data and green is mid range values. There are red cells at zoom 7 in this example
![image](https://user-images.githubusercontent.com/5776766/226642325-93b64f0d-229d-46d8-aa84-fe51217b1fe3.png)

When I zoom to zomething greater like z10, those edge effects between data/no data disappear
![image](https://user-images.githubusercontent.com/5776766/226642736-f5ee00eb-4531-4a26-b877-99ca99fed363.png)

I tile call to produce one of these looks something like this. 
https://titiler.xyz/cog/tiles/WebMercatorQuad/7/23/45@1x?url=urlToMyDatasetbidx=1&maxzoom=8&nodata=0&rescale=1,255&colormap_name=rdylgn

I was thinking changing it to a different interpolation like MIN would fix the issue but I am not allowed to use that method for some reason.

**Comment by vincentsarago** [comment, marked as answer]
My guess is that the artefact comes from the overviews creations. do you know what resampling you used when the COG was created @gagecarto

**Reply by gagecarto** [reply]
Bingo! 

Overviews were resampled with bilinear.. I'll try nearest and see if that clears things up.

Thanks @vincentsarago

**Reply by mradamcox** [reply]
@gagecarto I'm seeing some similar artifacts in my own images, which are JPEG-compressed COGs. Did you find a resampling algorithm that doesn't cause this?

**Reply by gagecarto** [reply]
@mradamcox - switching to nearest fixed it for me
